src/main/kotlin/net/fabricmc/loom/kotlin/remapping/KotlinClassMetadataRemappingAnnotationVisitor.kt

```diff
@@ -58,39 +58,40 @@ class KotlinClassMetadataRemappingAnnotationVisitor(private val remapper: Remapp
 
         when (val metadata = KotlinClassMetadata.read(header)) {
             is KotlinClassMetadata.Class -> {
-                var klass = metadata.toKmClass()
+                var klass = metadata.kmClass
                 klass = KotlinClassRemapper(remapper).remap(klass)
-                val remapped = KotlinClassMetadata.writeClass(klass, header.metadataVersion, header.extraInt).annotationData
+                val remapped = KotlinClassMetadata.writeClass(klass, header.metadataVersion, header.extraInt)
                 writeClassHeader(remapped)
                 validateKotlinClassHeader(remapped, header)
             }
             is KotlinClassMetadata.SyntheticClass -> {
-                var klambda = metadata.toKmLambda()
+                var klambda = metadata.kmLambda
 
                 if (klambda != null) {
                     klambda = KotlinClassRemapper(remapper).remap(klambda)
-                    val remapped = KotlinClassMetadata.writeLambda(klambda, header.metadataVersion, header.extraInt).annotationData
+                    val remapped = KotlinClassMetadata.writeLambda(klambda, header.metadataVersion, header.extraInt)
                     writeClassHeader(remapped)
                     validateKotlinClassHeader(remapped, header)
                 } else {
                     accept(next)
                 }
             }
             is KotlinClassMetadata.FileFacade -> {
-                var kpackage = metadata.toKmPackage()
+                var kpackage = metadata.kmPackage
                 kpackage = KotlinClassRemapper(remapper).remap(kpackage)
-                val remapped = KotlinClassMetadata.writeFileFacade(kpackage, header.metadataVersion, header.extraInt).annotationData
+                val remapped = KotlinClassMetadata.writeFileFacade(kpackage, header.metadataVersion, header.extraInt)
                 writeClassHeader(remapped)
                 validateKotlinClassHeader(remapped, header)
             }
             is KotlinClassMetadata.MultiFileClassPart -> {
-                var kpackage = metadata.toKmPackage()
+                var kpackage = metadata.kmPackage
                 kpackage = KotlinClassRemapper(remapper).remap(kpackage)
-                val remapped = KotlinClassMetadata.writeMultiFileClassPart(kpackage, metadata.facadeClassName, metadata.annotationData.metadataVersion, metadata.annotationData.extraInt).annotationData
+                val wrapper = KotlinClassMetadataWrapper(metadata)
+                val remapped = KotlinClassMetadata.writeMultiFileClassPart(kpackage, metadata.facadeClassName, wrapper.annotationData.metadataVersion, wrapper.annotationData.extraInt)
                 writeClassHeader(remapped)
                 validateKotlinClassHeader(remapped, header)
             }
-            is KotlinClassMetadata.MultiFileClassFacade, is KotlinClassMetadata.Unknown, null -> {
+            is KotlinClassMetadata.MultiFileClassFacade, is KotlinClassMetadata.Unknown -> {
```

Comment:
Note for future self:

As per the [migration guide](https://github.com/JetBrains/kotlin/blob/master/libraries/kotlinx-metadata/jvm/Migration.md#migrating-from-06x-to-070), we may want to try catch here.

Reply:
Going to leave this for now, better to crash and then figure out the error vs contiuning. If anyone has any issues with this please open an issue 👍